Implemented SQL Query Builder with clause components to generate rawSQL.

hiralpatelHM opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. To surface this PR in the changelog add the label: changelog
    If this PR is going in the changelog please make sure the title of the PR explains the feature in a user-centric way:
        Bad: fix state bug in hooks
        Good: Fix crash when switching from Query Builder

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, consider updating the documentation in README.md(https://github.com/grafana/iot-sitewise-datasource/blob/main/README.md) or [src/README.md](https://github.com/grafana/iot-sitewise-datasource/blob/main/README.md).

-->

**What this PR does / why we need it**:
This PR introduces the full SQL Query Builder module for Grafana panels. It enables users to build SQL queries through a dynamic UI that supports:
- Clause-based query construction (`SELECT`, `FROM`, `WHERE`, `GROUP BY`, `HAVING`, `ORDER BY`, `LIMIT`)
- Auto-generated raw SQL preview
- Function support with dynamic argument rendering (`date`, `cast`, `concat`,`str_replace`)
- State handling across component interactions
- Test coverage for components, hooks, and state logic

<img width="1573" height="597" alt="sql-builder" src="https://github.com/user-attachments/assets/5e3aa0c4-58d2-47bf-ba31-1b1d51cb1afc" />

This feature aims to improve usability for users who are not comfortable writing raw SQL, and makes query building more intuitive and robust.

**Which issue(s) this PR fixes**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer**:
- This feature is conditionally enabled — it becomes active only when the Code Editor is enabled in the panel options.
- This changes introduces a standalone SQL query builder module, not affecting existing modes.
- Please review `useSQLQueryState`, and `RawQueryEditor` closely for state flow

idastambuk commented:
Hi @hiralpatelHM thanks a lot for this contribution! 
Here's some initial feedback, mostly usability notes:

1. This adds a SQL builder to the query editor, but it's placed under the "code" query mode, which is a bit misleading. I think it makes more sense for this to be an additional "builder" mode, just like the builder we have right now.  (and it would have a corresponding "sql code" mode) Maybe we could have a dropdown to switch between two different builder (query) languages:
<img width="700" height="162" alt="Screenshot 2025-08-20 at 12 19 28" src="https://github.com/user-attachments/assets/479e763f-ee52-455c-871d-8c49d462afa6" />


We could get rid of the mode switcher then:
<img width="41" height="35" alt="Screenshot 2025-08-20 at 12 22 00" src="https://github.com/user-attachments/assets/689f0ef5-19f4-44cb-8e5f-ac7c50d4abd0" />

2. The validation error message is shown before the user has had a chance to build an sql query. Is it possible to only show errors once the user tries to run a query? 
3. Raw values/aggregate dropdown could also be a cascader: https://developers.grafana.com/ui/latest/index.html?path=/story/inputs-cascader--simple This way we wouldn't have as many items at the top level of the dropdown
4. Switching code -> builder is a bit buggy - if I edit the query in the code editor and switch to the builder, the query doesn't update (which is fine, I imagine building the editor values this was would be quite complicated) but then if I run the query, the query sent is from the Code editor and not the builder. This is quite confusing. I think the simplest thing would be to clear the query when the user switches Code -> sql builder. We could also display a warning popup telling the user that switching in this direction will clear the query. CloudWatch editor has this implemented:
<img width="600" height="384" alt="Screenshot 2025-08-20 at 12 25 41" src="https://github.com/user-attachments/assets/36b3f71b-47d3-4540-8822-dd562a7663ee" />

idastambuk commented on jest-setup.js:
```diff
@@ -1,32 +1,35 @@
 // Jest setup provided by Grafana scaffolding
 import './.config/jest-setup';
+// Polyfill IntersectionObserver for jsdom environment
+import 'intersection-observer';
```
We don't need an additional dependency, we can just mock this like:
```
// mock the intersection observer and just say everything is in view
const mockIntersectionObserver = jest.fn().mockImplementation(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
}));
global.IntersectionObserver = mockIntersectionObserver;
```

idastambuk commented on src/components/query/sql-query-builder/VariableSuggestInput.tsx:
```diff
@@ -0,0 +1,124 @@
+  };
+
+  /**
+   * Inserts selected suggestion into the input field at the last `$` position,
```
Why are we wrapping the variable in ${}?

Couldn't we use the [Basic Select with plain values](https://github.com/grafana/grafana/blob/566460f4fed7bc3388b937bd10708edbb854732f/packages/grafana-ui/src/components/Select/Select.story.tsx#L152) component here with allowCustomValue to true, if we wanted to display variables + allow the user to enter the value. We always try to use existing components from `grafana-ui` rather than custom inputs since they're more maintainable. Let us know if you need help implementing the Select here!

hiralpatelHM replied:
@idastambuk  changes pushed -  allowCustomValue true for the custom variable selection in where clause.

idastambuk commented on src/SitewiseDataSource.ts:
```diff
@@ -178,7 +178,7 @@ export class DataSource extends DataSourceWithBackend<SitewiseQuery, SitewiseOpt
         ? (templateSrv.replace(query.resolution, scopedVars) as SiteWiseResolution)
         : undefined,
-      rawSQL: templateSrv.replace(query.rawSQL, scopedVars),
+      rawSQL: templateSrv.replace(query.rawSQL, scopedVars, variableFormatter),
```
I think instead of the custom variableFormatter we can pass the 'sql' string for formatter arg ([this](https://github.com/grafana/scenes/blob/33a213f4dad54a35c51af33c95bc9be36f090846/packages/scenes/src/variables/interpolation/formatRegistry.ts#L424) is the formatting function it resolves to)

hiralpatelHM replied:
@idastambuk  changes pushed for the  'sqlstring' formatter of the RawSQL.

hiralpatelHM commented:
> > Hi @hiralpatelHM thanks a lot for this contribution! Here's some initial feedback, mostly usability notes:
> > 
> > 1. This adds a SQL builder to the query editor, but it's placed under the "code" query mode, which is a bit misleading. I think it makes more sense for this to be an additional "builder" mode, just like the builder we have right now.  (and it would have a corresponding "sql code" mode) Maybe we could have a dropdown to switch between two different builder (query) languages:
> > 
> > <img alt="Screenshot 2025-08-20 at 12 19 28" width="700" height="162" src="https://private-user-images.githubusercontent.com/16140639/479954262-479e763f-ee52-455c-871d-8c49d462afa6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU4Nzk0MTcsIm5iZiI6MTc1NTg3OTExNywicGF0aCI6Ii8xNjE0MDYzOS80Nzk5NTQyNjItNDc5ZTc2M2YtZWU1Mi00NTVjLTg3MWQtOGM0OWQ0NjJhZmE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIyVDE2MTE1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2ZWVlYWMwOTE5NjkzODkxYTgzN2M4NzE5YzdlMWQ2YWU0MWRiN2U2MDdlOTMwZjZmMTdiNWRkNzU0OTM5YmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.EY0BeyIHdv9ympaO1MJ9g1Fq-uYebS6XNEb-pPk9DiY">
> > We could get rid of the mode switcher then: <img alt="Screenshot 2025-08-20 at 12 22 00" width="41" height="35" src="https://private-user-images.githubusercontent.com/16140639/479952080-689f0ef5-19f4-44cb-8e5f-ac7c50d4abd0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU4Nzk0MTcsIm5iZiI6MTc1NTg3OTExNywicGF0aCI6Ii8xNjE0MDYzOS80Nzk5NTIwODAtNjg5ZjBlZjUtMTlmNC00NGNiLThlNWYtYWM3YzUwZDRhYmQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIyVDE2MTE1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBkODY2NDRmODk4YTU2NjBmNjM3NDJmMzI4MTY1N2FhM2M3ZmZkMTU4NjczYzUxYTUwOWM0YjAzYmU5MjkxZGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.29B4vad9S3bT9C9QaFb3WtOpNSH62MOqyuynFiOZmZA">
> > 
@idastambuk 
We can update the query editor by replacing the existing two-mode toggle with a dropdown selector for improved usability. The new dropdown will provide three clear options: Builder (the existing visual query builder), SQL Builder (a new structured SQL-based builder), and SQL Code (a raw SQL text editor). This requires UI changes. Please confirm whether we can proceed with this. 
> > 2. The validation error message is shown before the user has had a chance to build an sql query. Is it possible to only show errors once the user tries to run a query?
 Query preview is meant to help users by flagging missing required values early, before the query is sent to the API. That’s why the error appears right away, even before the query is executed.Should we have to keep the preview?
> > 3. Raw values/aggregate dropdown could also be a cascader: https://developers.grafana.com/ui/latest/index.html?path=/story/inputs-cascader--simple This way we wouldn't have as many items at the top level of the dropdown
> > 4. Switching code -> builder is a bit buggy - if I edit the query in the code editor and switch to the builder, the query doesn't update (which is fine, I imagine building the editor values this was would be quite complicated) but then if I run the query, the query sent is from the Code editor and not the builder. This is quite confusing. I think the simplest thing would be to clear the query when the user switches Code -> sql builder. We could also display a warning popup telling the user that switching in this direction will clear the query. CloudWatch editor has this implemented:
> > 
> > <img alt="Screenshot 2025-08-20 at 12 25 41" width="600" height="384" src="https://private-user-images.githubusercontent.com/16140639/479953564-36b3f71b-47d3-4540-8822-dd562a7663ee.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU4Nzk0MTcsIm5iZiI6MTc1NTg3OTExNywicGF0aCI6Ii8xNjE0MDYzOS80Nzk5NTM1NjQtMzZiM2Y3MWItNDdkMy00NTQwLTg4MjItZGQ1NjJhNzY2M2VlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIyVDE2MTE1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjYmZiZTYxZTM5Yjc4MWI4ZDMyYzczMTU5OGZkNDU1ZmYzZWFhNzA2NmZiZTc1ZjZkMjQ4OTc1MTA4NjY0ODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.l_lGcMmbOOIsZr1bTAmWmsA5JOyimKPDrQ1MVOfjsrc">
@idastambuk 
We can add a warning popup when switching from Code to SQL Builder. If the user confirms by clicking OK, the existing code editor data will be cleared.

hiralpatelHM commented:
> > > > Hi @hiralpatelHM thanks a lot for this contribution! Here's some initial feedback, mostly usability notes:
> > > > 
> > > > 1. This adds a SQL builder to the query editor, but it's placed under the "code" query mode, which is a bit misleading. I think it makes more sense for this to be an additional "builder" mode, just like the builder we have right now.  (and it would have a corresponding "sql code" mode) Maybe we could have a dropdown to switch between two different builder (query) languages:
> > > > 
> > > > <img alt="Screenshot 2025-08-20 at 12 19 28" width="700" height="162" src="https://private-user-images.githubusercontent.com/16140639/479954262-479e763f-ee52-455c-871d-8c49d462afa6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU4Nzk0MTcsIm5iZiI6MTc1NTg3OTExNywicGF0aCI6Ii8xNjE0MDYzOS80Nzk5NTQyNjItNDc5ZTc2M2YtZWU1Mi00NTVjLTg3MWQtOGM0OWQ0NjJhZmE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIyVDE2MTE1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2ZWVlYWMwOTE5NjkzODkxYTgzN2M4NzE5YzdlMWQ2YWU0MWRiN2U2MDdlOTMwZjZmMTdiNWRkNzU0OTM5YmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.EY0BeyIHdv9ympaO1MJ9g1Fq-uYebS6XNEb-pPk9DiY">
> > > > We could get rid of the mode switcher then: <img alt="Screenshot 2025-08-20 at 12 22 00" width="41" height="35" src="https://private-user-images.githubusercontent.com/16140639/479952080-689f0ef5-19f4-44cb-8e5f-ac7c50d4abd0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU4Nzk0MTcsIm5iZiI6MTc1NTg3OTExNywicGF0aCI6Ii8xNjE0MDYzOS80Nzk5NTIwODAtNjg5ZjBlZjUtMTlmNC00NGNiLThlNWYtYWM3YzUwZDRhYmQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIyVDE2MTE1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBkODY2NDRmODk4YTU2NjBmNjM3NDJmMzI4MTY1N2FhM2M3ZmZkMTU4NjczYzUxYTUwOWM0YjAzYmU5MjkxZGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.29B4vad9S3bT9C9QaFb3WtOpNSH62MOqyuynFiOZmZA">
> > 
> > 
> > @idastambuk We can update the query editor by replacing the existing two-mode toggle with a dropdown selector for improved usability. The new dropdown will provide three clear options: Builder (the existing visual query builder), SQL Builder (a new structured SQL-based builder), and SQL Code (a raw SQL text editor). This requires UI changes. Please confirm whether we can proceed with this.
> > > > 2. The validation error message is shown before the user has had a chance to build an sql query. Is it possible to only show errors once the user tries to run a query?
> >    @idastambuk  Query preview is meant to help users by flagging missing required values early, before the query is sent to the API. That’s why the error appears right away, even before the query is executed.Should we have to keep the preview?
> > > > 3. Raw values/aggregate dropdown could also be a cascader: https://developers.grafana.com/ui/latest/index.html?path=/story/inputs-cascader--simple This way we wouldn't have as many items at the top level of the dropdown
> > > > 4. Switching code -> builder is a bit buggy - if I edit the query in the code editor and switch to the builder, the query doesn't update (which is fine, I imagine building the editor values this was would be quite complicated) but then if I run the query, the query sent is from the Code editor and not the builder. This is quite confusing. I think the simplest thing would be to clear the query when the user switches Code -> sql builder. We could also display a warning popup telling the user that switching in this direction will clear the query. CloudWatch editor has this implemented:
> > > > 
> > > > <img alt="Screenshot 2025-08-20 at 12 25 41" width="600" height="384" src="https://private-user-images.githubusercontent.com/16140639/479953564-36b3f71b-47d3-4540-8822-dd562a7663ee.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU4Nzk0MTcsIm5iZiI6MTc1NTg3OTExNywicGF0aCI6Ii8xNjE0MDYzOS80Nzk5NTM1NjQtMzZiM2Y3MWItNDdkMy00NTQwLTg4MjItZGQ1NjJhNzY2M2VlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIyVDE2MTE1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjYmZiZTYxZTM5Yjc4MWI4ZDMyYzczMTU5OGZkNDU1ZmYzZWFhNzA2NmZiZTc1ZjZkMjQ4OTc1MTA4NjY0ODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.l_lGcMmbOOIsZr1bTAmWmsA5JOyimKPDrQ1MVOfjsrc">
> > 
> > 
> > @idastambuk We can add a warning popup when switching from Code to SQL Builder. If the user confirms by clicking OK, the existing code editor data will be cleared.

hiralpatelHM commented:
> Hi again @hiralpatelHM, great work so far. I added some remarks inline After testing the feature I noticed a bug when writing into the input fields:
> 
> 1. getting errors in the console when writing into Where clause inputs
> 2. maybe related to the errors - when writing in the input fields, there's a pretty significant lag (or a problem with the state management with the debounce function) where only the last letter of the word will be saved, unless I type really slowly

@idastambuk  I fixed the errors.

idastambuk commented on src/components/query/sql-query-builder/StyledLabel.tsx:
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
```
Hi @hiralpatelHM Im not sure I understand yet why we need the custom solutions and not use the same styling as in the CloudWatch component that I linked.For adding columns and group by clauses, for example - they can have `allowCustomValue` set to `true` if they're a dropdown that the user can add additional inputs to. But I might be missing something that prevents us from styling the form the same

hiralpatelHM commented on src/common/ConfirmDialog.tsx:
```diff
@@ -0,0 +1,29 @@
+// components/common/ConfirmDialog.tsx
```
@kevinwcyu  Replaced custom modal with ConfirmModal from @grafana/ui

hiralpatelHM commented on src/components/query/sql-query-builder/QueryPreviewDisplay.tsx:
```diff
@@ -0,0 +1,67 @@
+};
+
+// Style for each individual error message item
+const errorItemStyle: React.CSSProperties = {
+  fontSize: 12,
+};
+
+// Style for the query preview text block
+const previewStyle: React.CSSProperties = {
+  fontSize: 12,
+  padding: 0,
+  margin: 0,
+  backgroundColor: 'transparent',
+  border: 'none',
+  whiteSpace: 'pre-wrap',
+  wordBreak: 'break-word',
+};
```
@kevinwcyu 
Updated QueryPreviewDisplay component using @emotion/css.

hiralpatelHM commented on src/components/query/sql-query-builder/SqlQueryBuilder.tsx:
```diff
@@ -0,0 +1,99 @@
+  const isHavingVisible = queryState.groupByTags.length > 0;
+
+  return (
+    <div className="gf-form-group">
```
@kevinwcyu - Removed deprecated gf-form-group and applied @emotion/css for styling

hiralpatelHM commented on src/components/query/sql-query-builder/clauses/SelectClauseEditor.tsx:
```diff
@@ -0,0 +1,151 @@
+    updateQuery({ selectFields: newFields });
+  };
+
+  const removeSelectField = (index: number) => {
```
@kevinwcyu 
Fixed bug where removing a field did not correctly reset the aggregation value.

hiralpatelHM commented on src/SitewiseQueryEditor.tsx:
```diff
@@ -61,18 +123,37 @@ export function SitewiseQueryEditor(props: Props) {
-            query={query}
-            onChange={onChange}
-            onRunQuery={onRunQuery}
+          <CodeEditor
+            language="sql"
+            showLineNumbers
+            showMiniMap={false}
+            value={query.rawSQL || datasource.defaultQuery}
+            onSave={(text) => onChange({ ...query, rawSQL: text })}
+            onBlur={(text) => onChange({ ...query, rawSQL: text })}
+            onBeforeEditorMount={(monaco) => {
+              if (SitewiseCompletionProvider.monaco === null) {
+                SitewiseCompletionProvider.monaco = monaco;
+                monaco.languages.registerCompletionItemProvider('sql', SitewiseCompletionProvider);
+              }
+            }}
+            height={'200px'}
```
@kevinwcyu 
The main reason for inlining RawQueryEditor here is that **SitewiseQueryEditor**serves as the dropdown control for selecting and managing all three editor modes—Visual Query Builder, Code Editor, and SQL Builder. Keeping them together in one component ensures consistent state management, mode-switching logic, and query synchronization across all editors. It also makes easier to handle mode changes and update query data after a confirmation dialog action.

hiralpatelHM commented on src/components/query/sql-query-builder/clauses/FromClauseEditor.tsx:
```diff
@@ -0,0 +1,74 @@
+            err.type === 'from' && (
+              <div key={idx} className="text-error text-sm">
+                <div>{err.error}</div>
+              </div>
```
The attached screenshot shows how it would look if we used the **Alert** component. To match the error message style, I used the **Text** component with **variant="code"** and **color="error"** for displaying the validation errors

<img width="1489" height="466" alt="validation-error" src="https://github.com/user-attachments/assets/86162f34-4a83-41f3-b6df-c1961aaff942" />

hiralpatelHM commented on src/components/query/sql-query-builder/clauses/FromClauseEditor.tsx:
```diff
@@ -0,0 +1,74 @@
+  return (
+    <FieldSet label="From" style={{ marginBottom: 0 }}>
+      <Stack direction="row" gap={4} alignItems="center">
+        <EditorField label="View" width={40}>
```
@kevinwcyu Added the htmlFor and inputId for from component

hiralpatelHM commented:
@kevinwcyu 
- Replaced custom modal with ConfirmModal from @grafana/ui
- Updated query preview to use Text component instead of Alert
- Replaced all error messages with Text component for consistency
- Added htmlFor and inputId attributes in GroupBy and From components
- Removed initial load error from SQL Builder
- Updated marginBottom styling for From and Limit components

hiralpatelHM reviewed (commented):
@kevinwcyu

- Replaced custom modal with ConfirmModal from @grafana/ui
- Updated query preview to use Text component instead of Alert
- Replaced all error messages with Text component for consistency
- Added htmlFor and inputId attributes in GroupBy and From components
- Error in SQL Builder is now shown only after the user starts building a query
- Updated marginBottom styling for From and Limit components
